services/src/machine_learning/xgb_config_builder.rs

```diff
@@ -0,0 +1,1057 @@
+/// This module contains the configuration builder for xgboost.
+/// See [https://xgboost.readthedocs.io/en/stable/parameter.html](https://xgboost.readthedocs.io/en/stable/parameter.html) for more details on the parameters.
+/// The `BoosterParameters` struct is the endpoint, that is sent to xgboost.
+use derive_builder::Builder;
+use std::borrow::Borrow;
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::default::Default;
+use std::fmt::{self, Display};
+
+use serde::{Deserialize, Serialize};
+use xgboost_rs::DMatrix;
+
+use crate::error::Result;
+
+// -----------------------------------------------------------------------------------------------
+// XGBoost Config Definitions
+// -----------------------------------------------------------------------------------------------
+
+#[derive(Builder, Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+#[builder(default)]
+pub struct BoosterParameters {
+    booster_type: BoosterType,
+
+    learning_params: LearningTaskParameters,
+
+    verbose: bool,
+
+    threads: Option<u32>,
+}
+
+impl BoosterParameters {
+    pub fn booster_type(&self) -> &BoosterType {
```

Comment:
do you really need the getters and setters here? Isn't that what the derive_builder should generate automatically?